### PR TITLE
add rule-parameters to composites which include epsilon/gamma/alphabeta rule.

### PR DESCRIPTION
**Pull request by dkrako**
I thought it would be nice to have the epsilon parameter available for the composites that include the epsilon rule.

Is it necessary to include tests for this simple addition? Code test coverage shouldn't change at all with this addition.

### CODE REVIEW

**Comment by dkrako** [comment]
I now also added parameters for gamma, alpha and beta where it was used and documented all those newly added parameters.

**Comment by chr5tphr** [comment]
Hey Daniel,

thank you for the PR!

### Concerning the Design Choice
The built-in Composites are the most commonly used set of rules.
By Zennit's design of simplicity and customizability, I would rather like to keep them as simple as possible to encourage custom Composites using abstract Composites like LayerMapComposite instead when the use-case of changing them arises.

If you have not done so already, you may have a look at the [User Guide on Composites](https://zennit.readthedocs.io/en/latest/how-to/use-rules-composites-and-canonizers.html#composites), where examples are shown for defining a custom set of rules using `LayerMapComposite`, `SpecialFirstLayerMapComposite` and `NameMapCompposite`.

### Concerning the Use-Cases
Regarding AlphaBeta, it is not really common to choose anything except alpha=2, beta=1 and alpha=1, beta=0, so I think the alpha-beta parameters should be seen more as a static rule, thus I do not think this is a good addition.

Regarding Epsilon and Gamma, I think changing those may have actual use-cases, although I am not quite convinced yet given the customizability argument.
I know of one instance where the someone wanted to only change the gamma parameter in `EpsilonGammaBox`.

Did the change of the epsilon parameter arise in your research? Maybe if this is common enough, it may be indeed a good idea to introduce the epsilon parameter into the `__init__` of the Composites.

What are your thoughts and experiences as a user? Does it seem cumbersome to you to implement a custom composite only to change these parameters?

File: src/zennit/composites.py [hunk 1]
```diff
@@ -162,11 +162,15 @@ class EpsilonGammaBox(SpecialFirstLayerMapComposite):
         A tensor with the same size as the input, describing the lowest possible pixel values.
     high: obj:`torch.Tensor`
         A tensor with the same size as the input, describing the highest possible pixel values.
+    epsilon: float
+        Epsilon parameter for epsilon rule.
+    gamma: float
+        Gamma parameter for gamma rule.
```
Comment (chr5tphr) [inline src/zennit/composites.py:168]:
```suggestion
    epsilon: float
        Epsilon parameter for the epsilon rule.
    gamma: float
        Gamma parameter for the gamma rule.
```

File: src/zennit/composites.py [hunk 2]
```diff
@@ -162,11 +162,15 @@ class EpsilonGammaBox(SpecialFirstLayerMapComposite):
...
     '''
-    def __init__(self, low, high, canonizers=None):
+    def __init__(self, low, high, epsilon=1e-6, gamma=0.25, canonizers=None):
         layer_map = LAYER_MAP_BASE + [
-            (Convolution, Gamma(gamma=0.25)),
-            (torch.nn.Linear, Epsilon()),
+            (Convolution, Gamma(gamma=gamma)),
+            (torch.nn.Linear, Epsilon(epsilon)),
```
Comment (chr5tphr) [inline src/zennit/composites.py:173]:
```suggestion
            (torch.nn.Linear, Epsilon(epsilon=epsilon)),
```

File: src/zennit/composites.py [hunk 3]
```diff
@@ -178,60 +182,98 @@ def __init__(self, low, high, canonizers=None):
...
+class EpsilonAlpha2Beta1(EpsilonAlphaBeta):
+    '''An explicit composite using the alpha2-beta1 rule for all convolutional layers and the epsilon rule for all
+    fully connected layers.
+    epsilon: float
+        Epsilon parameter for epsilon rule.
+    '''
+    def __init__(self, epsilon=1e-6, canonizers=None):
+        super().__init__(alpha=2, beta=1, canonizers=canonizers)
+
+
```
Comment (chr5tphr) [inline src/zennit/composites.py:225]: Could you restore the original rule without the alpha, beta parameters, but with the epsilon?

Also use [numpydocstyle](https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring) so Sphinx can correctly parse the docstring.

File: src/zennit/composites.py [hunk 4]
```diff
@@ -178,60 +182,98 @@ def __init__(self, low, high, canonizers=None):
...
+class EpsilonAlpha2Beta1Flat(EpsilonAlphaBetaFlat):
+    '''An explicit composite using the flat rule for any linear first layer, the alpha2-beta1 rule for all other
+    convolutional layers and the epsilon rule for all other fully connected layers.
+    epsilon: float
+        Epsilon parameter for epsilon rule.
+    '''
+    def __init__(self, epsilon=1e-6, canonizers=None):
+        super().__init__(alpha=2, beta=1, canonizers=canonizers)
+
+
```
Comment (chr5tphr) [inline src/zennit/composites.py:276]: Could you restore the original rule without the alpha, beta parameters, but with the epsilon?

Also use [numpydocstyle](https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring) so Sphinx can correctly parse the docstring.

**Review by chr5tphr — CHANGES_REQUESTED** [review]
Thanks a lot!

I decided to keep the epsilon and gamma parameters, but throw out the alpha and beta parameters.

Pylint was previously failing due to an update, I have fixed this in master, so you can rebase to get rid of those issues.

The docs were failing because you did not use [numpydocstyle](https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring) for the parameters.

Could you also, after you are done, squash your commits and format them like the following:

```
Composites: Short summary text

- first change and explanation
- another change and explanation
```

Thanks a lot for your contribution!

**Pull request merged**